File: backend/src/main/java/com/fabriciodev/model/Consulta.java
package com.fabriciodev.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "consultas")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Consulta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "paciente_id")
    private Paciente paciente;

    @ManyToOne
    @JoinColumn(name = "medico_id")
    private Medico medico;

    private LocalDateTime dataConsulta;
    private String status;
}

File: frontend/src/app/web/interfaces/consulta.ts
import { Medico } from "./medico";
import { Paciente } from "./paciente";

export interface Consulta {
  id: number;
  paciente_id: number;
  medico_id: number;
  dataConsulta: Date;
  status: string;

  paciente: Paciente;
  medico: Medico;
}
